roachtest: disable create_table_with_schema_locked for certain tests

fqazi opened the pull request:
This series of commits addresses several test failures across different roachtest suites (`costfuzz`, `unoptimized-query-oracle`, `follower-reads`, and `sqlsmith`) that were caused by the `schema_locked` table setting. The multi-region version of these tests preform `ALTER TABLE ... SET LOCALITY`, so these have been modified either to toggle schema_locked or disable it for newly created tables.

This patch also fixes a flake inside TestChangefeedNemeses by disabling create_table_with_schema_locked when the legacy schema changer is in use.

Fixes: #148927
Fixes: #148940
Fixes: #149007
Fixes: #148976
Fixes: #148975
Fixes: #149002
Fixes: #148994
Fixes: #148995
Fixes: #149005
Fixes: #149000
Fixes: #148999
Fixes: #148996
Fixes: #149018
Fixes: #149015
Fixes: #149013
Fixes: #148997
Fixes: #149047
Fixes: #149045
Fixes: #149017
Fixes: #149014
Fixes: #149019
Fixes: #149016
Fixes: #149012

Release note: None

fqazi committed:
roachtest: disable schema locked for follower-reads

Previously, this test used ALTER TABLE .. SET LOCALITY which is not
supported without manually unsetting schema_locked. To address this,
this patch disables schema_locked for new tables within this test.

Fixes: #148996
Fixes: #149018
Fixes: #149015
Fixes: #149013
Fixes: #148997
Fixes: #149047
Fixes: #149045
Fixes: #149017
Fixes: #149014
Fixes: #149019
Fixes: #149016
Fixes: #149012

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149009)
<!-- Reviewable:end -->

rafiss commented:
Does this cover all the tests that @pav-kv linked in his search: https://github.com/cockroachdb/cockroach/issues/148996#issuecomment-3012668626 ? It doesn't seem like it does.

fqazi committed:
roachtest: adjust costfuzz / unoptimized-query-oracle for schema_locked

Previously, the multi-region variants of costfuzz and
unoptimized-query-oracle, and sqlsmith would fail because they would
attempt to alter locality on locked tables. This patch disables
schema_locked before adjusting locality.

Fixes: #149007
Fixes: #148976
Fixes: #148975
Fixes: #149002
Fixes: #148994
Fixes: #148995
Fixes: #149005
Fixes: #149000
Fixes: #148999

Release note: None

fqazi committed:
ccl/changefeedccl: deflake TestChangefeedNemeses

Previously, this test would fail when the legacy schema changer was used
because tables would be created with schema_locked. This patch avoids
creating tables with schema_locked if the legacy scheam chagner is being
test.

Fixes: #148927
Fixes: #148940

Release note: None

rafiss reviewed (approved):
thanks for tracking down the fallout!

fqazi commented:
@rafiss TFTR!

bors r+
